### PR TITLE
Fix I2C issues with Raspberry Pi 2

### PR DESCRIPTION
**Pull request by penguineer**
See https://www.netz39.de/wiki/projects:2020:i2c_foo for details.

tl;dr:
* Raspberry Pi 2 (or rather: up-to-date i2c drivers) implement repeated starts
* USITWI library uses stop condition, which is now missing, to determine when the callback should be executed
* There is no generic marker between sending and receiving bytes in a request
* callback was only called _after_ the complete request, thus the result was in the buffer for the next request and this weird shift happened
*
Current solution: 
* We are only ever sending one request byte, so start processing as soon as there is one byte available.

The only generic marker would be to check the I2C direction bit, but that is sent _last_ in the address, so we loose the time the bus takes to send 8 address bytes for sending. (This was the clever part in this library. Why the master is sending the 8 address bytes to get the result, the slave had time for processing.)

This solution works for our I3C bus, but in general has room for improvement outside the scope of this PR

### CODE REVIEW

**Comment by penguineer** [comment]
@24367dfa
